### ISSUE TITLE
MNIST accuracy is bad

### ISSUE THREAD

**Issue by astorfi** [facebookresearch/pytorch-dp#7]
Hi,

When I run [mnist example](https://github.com/facebookresearch/pytorch-dp/blob/master/examples/mnist.py), the final accuracy is as below:

`Accuracy: 0.26980000734329224 ± nan`

Why it is so bad?

Thank you

**Comment by Darktex** [comment]
Can you share how you launched your experiment?

**Comment by Darktex** [comment]
These are example parameters that you can try: https://github.com/tensorflow/privacy/tree/master/tutorials#select-parameters

You should get a repro for each one of them in the table

**Comment by astorfi** [comment]
I have tried the following and it gives me:
based on the table of [the link you shared](https://github.com/tensorflow/privacy/tree/master/tutorials#select-parameters): 0.15 | 1.1 | 1.0 | 256 | 60

`0.8963000178337097 ± nan`

My issue is the `NaN` as well. Is it computed correctly?

**Comment by ilyamironov** [comment]
I've been able to reproduced the problem. Thank you for reporting it!

**Comment by astorfi** [comment]
@ilyamironov Thank you so much. I am looking forward to seeing why this is happening.

**Comment by astorfi** [comment]
@ilyamironov I am close to understanding why we get those NaN. Should I report here or make a pull request? I am doing this as I understand you might be very busy with a lot of responsibilities. Thank you.

**Comment by astorfi** [comment]
@ilyamironov I found the error.
According to [PyTorch docs](https://pytorch.org/docs/stable/torch.html#torch.std), the default provides an unbiased estimate for the std. This means when we run our experiments for only one time (n_runs==1), then the denominator of the std calculation would be zero (``n-1`` in unbiassed estimator). This is because the accuracy is calculated for each run and stored. For a higher number of runs, this is not an issue.

**Comment by ilyamironov** [comment]
The issue is addressed with commit https://github.com/facebookresearch/pytorch-dp/commit/1e925fd78a6a0fdc07428364c656c90a81c5391e
